### PR TITLE
restore a few new() in entities, refactor createTimelineHandler, add-…

### PR DESCRIPTION
**Pull request by metrondir**

### CODE REVIEW

**Review by Shavronskyy — APPROVED** [review]
lgtm

File: Streetcode/Streetcode.DAL/Persistence/Migrations/StreetcodeDbContextModelSnapshot.cs [hunk 1]
```diff
@@ -1553,14 +1556,12 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
             modelBuilder.Entity("Streetcode.DAL.Entities.Toponyms.Toponym", b =>
                 {
-                    b.Navigation("Coordinate")
-                        .IsRequired();
+                    b.Navigation("Coordinate");
                 });
```
Comment (serhiiyasenev) [inline Streetcode/Streetcode.DAL/Persistence/Migrations/StreetcodeDbContextModelSnapshot.cs:1560]: should these 2 fields be `Required` or not anymore?

File: Streetcode/Streetcode.DAL/Persistence/Configurations/Timeline/HistoricalContextConfiguration.cs [hunk 2]
```diff
@@ -17,6 +17,8 @@ public void Configure(EntityTypeBuilder<HistoricalContext> builder)
             builder.Property(h => h.Title)
                 .IsRequired()
                 .HasMaxLength(50);
+            
+            builder.HasIndex(h => h.Title).IsUnique();
```
Comment (serhiiyasenev) [inline Streetcode/Streetcode.DAL/Persistence/Configurations/Timeline/HistoricalContextConfiguration.cs:21]: why `Title` should be `Unique`?

File: Streetcode/Streetcode.BLL/MediatR/Timeline/TimelineItem/Create/CreateTimelineItemHandler.cs [hunk 3]
```diff
@@ -40,31 +39,33 @@ public async Task<Result<CreateTimelineItemDTO>> Handle(CreateTimelineItemComman
...
+                {
+                    HistoricalContextId = hc.Id,
+                    HistoricalContext = hc,
+                    TimelineId = newTimelineItem.Id,
+                    Timeline = newTimelineItem
+                }).ToList();
 
+                newTimelineItem.HistoricalContextTimelines.AddRange(historicalContextTimelines);
                 await _repositoryWrapper.SaveChangesAsync();
+
```
Comment (serhiiyasenev) [inline Streetcode/Streetcode.BLL/MediatR/Timeline/TimelineItem/Create/CreateTimelineItemHandler.cs:68]: we have 3 times `SaveChangesAsync();` - can we leave just 1?

and probably use `using var transaction = await _repositoryWrapper.BeginTransactionAsync();`  and if exception `await transaction.RollbackAsync();` 

it's not obligatory - just nice to have if possible

File: Streetcode/Streetcode.DAL/Persistence/Configurations/Timeline/HistoricalContextConfiguration.cs (hunk 2 above)
Comment (metrondir) [reply Streetcode/Streetcode.DAL/Persistence/Configurations/Timeline/HistoricalContextConfiguration.cs:21]: Cause actually cant exist two the same history context with the same title

File: Streetcode/Streetcode.DAL/Persistence/Migrations/StreetcodeDbContextModelSnapshot.cs (hunk 1 above)
Comment (metrondir) [reply Streetcode/Streetcode.DAL/Persistence/Migrations/StreetcodeDbContextModelSnapshot.cs:1560]: its snapshot i cant handle that

**Review by Forestieur — APPROVED** [review]

**Review by Shavronskyy — APPROVED** [review]
LGTM

**Pull request merged**